rllib/core/rl_module/rl_module.py

```diff
@@ -390,13 +402,88 @@ def get_inference_action_dist_cls(self) -> Type[Distribution]:
         """
         raise NotImplementedError
 
+    @OverrideToImplementCustomLogic
     def get_initial_state(self) -> NestedDict:
         """Returns the initial state of the module.
 
         This is used for recurrent models.
         """
         return {}
 
+    @OverrideToImplementCustomLogic
+    def is_recurrent(self) -> bool:
+        """Returns True if the initial state is empty.
+
+        By default, RLlib assumes that the module is not recurrent if the initial
+        state is an empty dict and recurrent otherwise.
+        This behavior can be overridden by implementing this method.
+        """
+        initial_state = self.get_initial_state()
+        if initial_state == {}:
+            return False
+        else:
+            return True
```

Comment:
I think this method is more elegant and offers better degrees of freedom for the user than us checking `RLModule.get_initial_state()` everytime we want to find out if the we are handling the recurrent case.